### ISSUE TITLE
Stickybombs sometimes shoot in a weird direction

### ISSUE THREAD

**Issue by braem** [momentum-mod/game#1059]
**Describe the bug**
Rarely, a stickybomb will shoot at an odd angle, almost as if it immediately bounces off something. The angle is reproducible if the player is in the same position and facing at the same angle.

Checking `VPhysicsCollision` in `mom_stickybomb` shows that the stickybomb is not bouncing off something, rather it is being shot at a weird angle.

This can easily kill runs.

**To Reproduce**
Difficult to reproduce as it's rare and you have to find the same position/angle which is occurs. 

A position/angle I found it to happen is:
`setpos 337.118622 -743.968750 -59.968750;setang 34.984016 26.188278 0.000000` on `sj_isak`.
also
`setpos 3077.678955 -7034.578125 -6075.968750;setang 80.344101 -152.367874 0.000000` on `sj_lanc`
`setpos 3073.657959 -7168.551758 -6075.968750;setang 81.092003 -179.075958 0.000000` on `sj_lanc`

More egregiously are the times where the sticky flies 90 degrees to the side.

**Expected behavior**
Stickybombs should always shoot forwards as normal, not sideways or backwards.

**Desktop/Branch (please complete the following information):**
 - OS: Windows
 - Branch: Steam version & Latest develop branch

**Comment by Gocnak** [comment]
I believe this happens due to the way that rollers are implemented. The func_physbox method, when used as a volume, can cause this effect when firing right on the edge of it, which is what's happening here.

https://developer.valvesoftware.com/wiki/Func_physbox

**Comment by braem** [comment]
> 
> 
> I believe this happens due to the way that rollers are implemented. The func_physbox method, when used as a volume, can cause this effect when firing right on the edge of it, which is what's happening here.
> 
> https://developer.valvesoftware.com/wiki/Func_physbox

Stickies are `func_physbox`'s? When it's rolling it still calls `VPhysicsCollide` though. When this bug happens it doesnt call that which lead me to believe it's being shot at a weird angle.

**Comment by Gocnak** [comment]
Sorry I meant to elaborate more:

That func entity, when used as a box (volume) to prevent stickies from sticking, and you "cut" your sticky launcher just right on it, causes this. It's like the sticky "rolls" off of it in mid air. 

Something to look at is the shoot angle for the gun but if that's not changing either then it's something with how vphysics objects handle this problem and might not be solvable.

Does TF2 have this problem?

**Comment by hexaflexahexagon** [comment]
> Does TF2 have this problem?

yes, and the setpos on sj_isak above shoots strangely in both momentum and tf2.

**Comment by braem** [comment]
Seems to be no clip vphysics on isak and also no `func_brush`es close to the player:

![image](https://user-images.githubusercontent.com/9014762/97096948-8c1ac680-1628-11eb-8f8d-be6d9d770180.png)

**Comment by hexaflexahexagon** [comment]
Not sure how helpful these will be with debugging causation but I've anecdotally noticed these facts about this bug, all on steam version 0.8.6: 
1. only happens when looking downwards
2. only happens when moving relatively slowly (from 0u to something slightly above walking speed, maybe ~500u?)
3. happens both mid-air and on the ground
4. happens both standing and crouched
5. charging does not affect the bug (happens regardless if set up at a reliable angle and coordinate)
6. stickies seem to randomly (1/6ish) not be affected by the bug in an otherwise consistent setup (I used `setpos 3077.877197 -7030.142090 -6075.968750;setang 79.662895 -153.758026 0.000000` on sj_lanc)
7. precise setups do not seem to matter and have a margin of error of a decent few hammer units / degrees. (i was not in mapping and recreated the setups in the original issue manually with `cl_showpos 1` and they worked very quickly, i was either lucky or the bug isnt super precise to replicate)
8. position and angle both are related (pos 1 2 3, angle 10 20 0 will work but pos 100 200 3, angle 10 20 0 most likely wont)

**Comment by Gocnak** [comment]
This was investigated to be a problem specifically with these maps, and deep in VPhysics code that we do not have.

Two solutions:
1. Repair the maps
2. Get VPhysics code and fix the underlying issue

We may just opt to go for 1, as I do not think we can do 2. I'll close this issue for now, if we somehow get VPhysics code we can re-open and try to fix it properly.

**Comment by hexaflexahexagon** [comment]
how exactly are mappers supposed to spot and fix this issue? i dont remember all the discourse around this but having that info explicit here would be very nice if that's the approach we take

**Comment by Gocnak** [comment]
For isak it was selecting all of the vphysics clips and making them individual instead of grouped iirc. Like you could select one and a bunch also get selected; breaking that up and making them individual helps.

**Comment by braem** [comment]
Yeah the issue seems to happen with `func_brush`s that have many brushes.
